pkg/wallet/box.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+import (
+	"errors"
+	"io"
+
+	"github.com/hyperledger/aries-framework-go/pkg/internal/cryptoutil"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/nacl/box"
+)
+
+// Box provides an elliptic-curve-based authenticated encryption scheme
```

Comment:
So I assume these are all wrappers of the box methods that need access to private material. One thing that confused me a bit was the method names don’t quite match the underlying wrapped box method.